### PR TITLE
Provide libwpe for webkitgtk in Wayland

### PR DESCRIPTION
**Pull request by matthewbauer**
When running in a Wayland environment, webkitgtk needs the libwpe backend to implement accelerated compositing on Wayland. Without this, you can get errors like:

>  Cannot get default EGL display: EGL_BAD_PARAMETER

Webkitgtk is supposed to gracefully disable AC when this happens, but this doesn't appear to be working for me. You can manually disable AC with WEBKIT_DISABLE_COMPOSITING_MODE=1.

See https://gitweb.gentoo.org/repo/gentoo.git/tree/net-libs/webkit-gtk/webkit-gtk-2.28.4.ebuild#n222 for some idea of how Gentoo handles this. In nixpkgs, I think it's a good idea to just provide libwpe everywhere, regardless of whether the user has configured "wayland" or not.

Other discussions:

- https://github.com/reflex-frp/reflex-platform/issues/691
- https://bugs.webkit.org/show_bug.cgi?id=202362
- https://discourse.nixos.org/t/install-nixos-on-laptop-im-failing-hard/6955

### CODE REVIEW

File: pkgs/development/libraries/libwpe/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version = "1.7.1";
+
+  src = fetchurl {
+    url = "https://wpewebkit.org/releases/${pname}-${version}.tar.xz";
+    sha256 = "0h6kh8wy2b370y705pl2vp6vp18dkdsgdxh0243ji2v51kxbg157";
+  };
+
+  buildInputs = [ libxkbcommon libGL libX11 ];
+
+  nativeBuildInputs = [ pkgconfig meson ninja ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libwpe/default.nix:14]: In new expressions, `pkg-config` should be used instead of the `pkgconfig` alias.

**Review by jtojnar — APPROVED** [review]
Weird, I thought we had already wpewebkit PR but cannot find it.

**Comment by matthewbauer** [comment]
> Weird, I thought we had already wpewebkit PR but cannot find it.

Ah I had opened it in https://github.com/NixOS/nixpkgs/pull/96140 but forgot about it! Anyway this should be find to merge first.

File: pkgs/development/libraries/libwpe/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libwpe/default.nix:14]: Please switch to pkg-config.

File: pkgs/development/libraries/libwpe/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with stdenv.lib; {
+    description = "General-purpose library for WPE WebKit";
+    license = licenses.bsd2;
+    homepage = "https://wpewebkit.org";
+    maintainers = [ maintainers.matthewbauer ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libwpe/default.nix:20]:
```suggestion
    maintainers = with maintainers; [ matthewbauer ];
```

File: pkgs/development/libraries/libwpe/fdo.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  depsBuildBuild = [
+    pkgconfig
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Freedesktop.org backend for WPE WebKit";
+    license = licenses.bsd2;
+    homepage = "https://wpewebkit.org";
+    maintainers = [ maintainers.matthewbauer ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libwpe/fdo.nix:48]:
```suggestion
    maintainers = with maintainers; [ matthewbauer ];
```

File: pkgs/development/libraries/libwpe/fdo.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  nativeBuildInputs = [
+    pkgconfig
+    meson
+    ninja
+    wayland
+  ];
+
+  depsBuildBuild = [
+    pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libwpe/fdo.nix:41]:
```suggestion
    pkg-config
```

File: pkgs/development/libraries/libwpe/fdo.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    wayland
+    epoxy
+    glib
+    libwpe
+    libxkbcommon
+    libGL
+    libX11
+  ];
+
+  nativeBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libwpe/fdo.nix:33]: Please list nativeBuildInputs before buildInputs.

**Comment by matthewbauer** [comment]
Updated

**Comment by matthewbauer** [comment]
@GrahamcOfBorg build epiphany

File: pkgs/development/libraries/libwpe/fdo.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    url = "https://wpewebkit.org/releases/${pname}-${version}.tar.xz";
+    sha256 = "1xf6akagvpyh0nyxkfijrx5avp6ravnivy28dhk64dsfx9rhm64v";
+  };
+
+  depsBuildBuild = [
+    pkg-config
+  ];
+
+  nativeBuildInputs = [
+    pkgconfig
```
Comment (jtojnar) [inline pkgs/development/libraries/libwpe/fdo.nix:28]: eval error

**Comment by SuperSandro2000** [comment]
@matthewbauer Can you fix the eval error?

**Comment by matthewbauer** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/96140

**Pull request closed without merging**